code/ui/ChatBox.cs

```diff
@@ -0,0 +1,137 @@
+using Sandbox;
+using Sandbox.UI;
+using Sandbox.UI.Construct;
+using System;
+using TTTReborn.Player;
+
+// Chat needs to hook into s&box in order to function.
+namespace Sandbox.Hooks
+{
+    public static partial class Chat
+    {
+        public static event Action OnOpenChat;
+
+        [ClientCmd("openchat")]
+        internal static void MessageMode()
+        {
+            OnOpenChat?.Invoke();
+        }
+
+    }
+}
+
+namespace TTTReborn.UI
+{
+    public partial class ChatBox : Panel
+    {
+        public static ChatBox Instance;
+
+        private readonly Panel _canvas;
+        private readonly TextEntry _input;
+
+        public ChatBox()
+        {
+            Instance = this;
+
+            StyleSheet.Load("/ui/ChatBox.scss");
+
+            _canvas = Add.Panel("chat_canvas");
+
+            _input = Add.TextEntry("");
+            _input.AddEventListener("onsubmit", Submit);
+            _input.AddEventListener("onblur", Close);
+            _input.AcceptsFocus = true;
+            _input.AllowEmojiReplace = true;
+
+            Sandbox.Hooks.Chat.OnOpenChat += Open;
+        }
+
+        private void Open()
+        {
+            AddClass("open");
+            _input.Focus();
+        }
+
+        private void Close()
+        {
+            RemoveClass("open");
+            _input.Blur();
+        }
+
+        private void Submit()
+        {
+            Close();
+
+            if (Local.Pawn is not TTTPlayer player)
+            {
+                return;
+            }
+
+            var msg = _input.Text.Trim();
+            _input.Text = "";
+
+            if (string.IsNullOrWhiteSpace(msg))
+                return;
+
+
+            Say(msg, player.LifeState);
+        }
+
+        public void AddEntry(string name, string message, string avatar, LifeState lifeState)
```

Comment:
Maybe we should pass the class name directly as it would be nice if traitors would see other traitors as well. Or an option adding additional classes.